src/providers/data_provider/dp_request.c

```diff
@@ -272,6 +283,8 @@ file_dp_request(TALLOC_CTX *mem_ctx,
     ret = EOK;
 
 done:
+    /* Restore the chain id to its original value when leaving this request. */
+    sss_chain_id_set(old_chain_id);
```

Comment:
Why it is done here and not at the end of `dp_attach_req()`? To have proper debug during `dp_req_new()`?

Reply:
Yes, exactly.